src/main/scala/db/migration/V15__NotesWithUserAndTimestamp.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Part of NDLA draft-api.
+ * Copyright (C) 2018 NDLA
+ *
+ * See LICENSE
+ */
+
+package db.migration
+
+import java.util.Date
+
+import no.ndla.draftapi.model.domain._
+import org.flywaydb.core.api.migration.{BaseJavaMigration, Context}
+import org.json4s.JsonAST.JArray
+import org.json4s.ext.EnumNameSerializer
+import org.json4s.native.JsonMethods.{compact, parse, render}
+import org.json4s.{Extraction, Formats}
+import org.postgresql.util.PGobject
+import scalikejdbc.{DB, DBSession, _}
+
+import scala.util.{Success, Try}
+
+class V15__NotesWithUserAndTimestamp extends BaseJavaMigration {
+  implicit val formats: Formats = org.json4s.DefaultFormats + new EnumNameSerializer(ArticleStatus)
+
+  override def migrate(context: Context): Unit = {
+    val db = DB(context.getConnection)
+    db.autoClose(false)
+
+    db.withinTx { implicit session =>
+      migrateArticleNotes
+    }
+  }
+
+  def migrateArticleNotes(implicit session: DBSession): Unit = {
+    val count = countAllArticles.get
+    var numPagesLeft = (count / 1000) + 1
+    var offset = 0L
+
+    while (numPagesLeft > 0) {
+      allArticles(offset * 1000).map {
+        case (id, document) => updateArticle(convertNotes(document), id)
+      }
+      numPagesLeft -= 1
+      offset += 1
+    }
+  }
+
+  def countAllArticles(implicit session: DBSession): Option[Long] = {
+    sql"select count(*) from articledata where document is not NULL"
+      .map(rs => rs.long("count"))
+      .single()
+      .apply()
+  }
+
+  def allArticles(offset: Long)(implicit session: DBSession): Seq[(Long, String)] = {
+    sql"select id, document from articledata where document is not null order by id limit 1000 offset $offset"
+      .map(rs => {
+        (rs.long("id"), rs.string("document"))
+      })
+      .list
+      .apply()
+  }
+
+  def convertNotes(document: String): String = {
+    val oldArticle = parse(document)
+    Try(oldArticle.extract[V14__Article]) match {
+      case Success(old) =>
+        val newArticle = oldArticle.mapField {
+          case ("notes", _) =>
+            val editorNotes = Seq(
+              EditorNote(
+                old.notes,
```

Comment:
En EditorNote per gammel note tenkte eg. Du får eventuellt sette updatedBy til "System" eller nokke i og med at det kan være fleire notes.

Reply:
Kan jeg ikke bare gjøre som jeg gjør nå (Beholde updatedBy på artiklen og sette user på notes til updatedBy ), bare at jeg dupliserer objektet pr note istedet for å kunne ha flere notes i hvert objekt?

Reply:
Dersom det er fleire notes veit du jo ikkje kven som har skrevet noten. Og eg meinte updatedBy på noten kan være System. Treng ikkje røre updatedBy på draften.